packages/cli/src/services/webhook.service.ts

```diff
@@ -0,0 +1,127 @@
+import { WebhookRepository } from '@/databases/repositories';
+import { Service } from 'typedi';
+import { CacheService } from './cache.service';
+import type { WebhookEntity } from '@/databases/entities/WebhookEntity';
+import type { IHttpRequestMethods } from 'n8n-workflow';
+import type { DeepPartial } from 'typeorm';
+
+type Method = NonNullable<IHttpRequestMethods>;
+
+@Service()
+export class WebhookService {
+	constructor(
+		private webhookRepository: WebhookRepository,
+		private cacheService: CacheService,
+	) {}
+
+	async populateCache() {
+		const allWebhooks = await this.webhookRepository.find();
+
+		if (!allWebhooks) return;
+
+		void this.cacheService.setMany(allWebhooks.map((w) => [w.cacheKey, w]));
+	}
+
+	private async findCached(method: Method, path: string) {
+		const cacheKey = `cache:webhook:${method}-${path}`;
+
+		const cachedWebhook = await this.cacheService.get(cacheKey);
```

Comment:
this block could now be turned into a single call (but still works this way as well of course)

Reply:
You mean combining these two lines? That's not possible because later `void this.cacheService.set(cacheKey, dbWebhook);` needs the `cacheKey`.